src/systems/two-point-stretching-system.js

```diff
@@ -0,0 +1,55 @@
+/* global THREE AFRAME */
+const distanceBetweenStretchers = (() => {
+  const a = new THREE.Vector3();
+  const b = new THREE.Vector3();
+  return function distanceBetweenStretchers(obj1, obj2) {
+    obj1.getWorldPosition(a);
+    obj2.getWorldPosition(b);
+    return a.distanceTo(b);
+  };
+})();
+
+export class TwoPointStretchingSystem {
+  constructor() {
+    this.heldLeftHand = null;
+    this.heldRightHand = null;
+    this.heldRightRemote = null;
+    this.initialScale = new THREE.Vector3();
+  }
+
+  tick() {
+    const interaction = AFRAME.scenes[0].systems.interaction;
+    const { leftHand, rightHand, rightRemote } = interaction.state;
+
+    const stretching = leftHand.held && (leftHand.held === rightHand.held || leftHand.held === rightRemote.held);
+    let stretcherLeft, stretcherRight;
+    if (stretching) {
+      stretcherLeft = interaction.options.leftHand.entity.object3D;
+      stretcherRight =
+        leftHand.held === rightHand.held
+          ? interaction.options.rightHand.entity.object3D
+          : interaction.options.rightRemote.entity.object3D;
+      if (
+        this.stretcherLeft !== stretcherLeft ||
+        this.stretcherRight !== stretcherRight ||
+        leftHand.held !== this.stretched
+      ) {
+        this.initialStretchDistance = distanceBetweenStretchers(stretcherLeft, stretcherRight);
+        this.stretched = leftHand.held;
+        this.initialScale.copy(this.stretched.object3D.scale);
+      }
+
+      this.stretched.object3D.scale
+        .copy(this.initialScale)
+        .multiplyScalar(distanceBetweenStretchers(stretcherLeft, stretcherRight) / this.initialStretchDistance);
```

Comment:
Marshall suggested that the amount you stretch an object should be such that the two points you begin touching are still the same two points on the model after you're done stretching. I suspect that would feel even better than this, but I'm not sure how to do that.